src/components/Breadcrumbs/BreadcrumbsItem.tsx

```diff
@@ -3,43 +3,43 @@ import {Link} from '../Link';
 import {block} from '../utils/cn';
 import type {BreadcrumbsItem as IBreadcrumbsItem, BreadcrumbsProps} from './Breadcrumbs';
 
-interface Props {
-    data: IBreadcrumbsItem;
+interface Props<T extends IBreadcrumbsItem = IBreadcrumbsItem> {
+    data: T;
     isCurrent: boolean;
     isPrevCurrent: boolean;
-    renderItem?: BreadcrumbsProps['renderItemContent'] | BreadcrumbsProps['renderRootContent'];
+    renderItem?:
+        | BreadcrumbsProps<T>['renderItemContent']
+        | BreadcrumbsProps<T>['renderRootContent'];
 }
 
 const b = block('breadcrumbs');
 
-export const BreadcrumbsItem = React.memo(function BreadcrumbsItem({
-    data,
-    isCurrent,
-    isPrevCurrent,
-    renderItem,
-}: Props) {
-    const {text, href, action} = data;
+export class BreadcrumbsItem<
+    T extends IBreadcrumbsItem = IBreadcrumbsItem,
+> extends React.PureComponent<Props<T>> {
+    render() {
```

Comment:
Why you convert FC component to class?

Reply:
It is impossible to create generic components with React.memo

Reply:
It is possible https://stackoverflow.com/a/60170425